### PR TITLE
MetallicityPDF: confidence intervals

### PR DESCRIPTION
**Pull request by profxj**
Introduces the MetallicityPDF class
Addresses Issue #7
test and docs and notebook too

### CODE REVIEW

**Comment by profxj** [comment]
I got myself confused on PR's.  I thought I had issued
one for MetallicityPDF already.  This is now the main one.

@mifumagalli , please inspect it with JMO.

File: pyigm/metallicity/pdf.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        Parameters
+        ----------
+        ZH : ndarray
+          Values of ZH (log10, typically but not required) where the PDF has been evaluated
+        pdf_ZH : ndarray
+          PDF of the metallicity
+        parent : object, optional
+          Parent of the metalliicty PDF, e.g. an AbsSystem
+        dZH : float or ndarray
+          Width of the ZH 'bins' (log 10)
```
Comment (mifumagalli) [inline pyigm/metallicity/pdf.py]: Note that must be regularly spaced, given line 49

File: pyigm/metallicity/pdf.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        frac = (1.- cl) / 2.
+        ZH_min = float(f(frac))
+        ZH_max = float(f(1-frac))
+        # Return
+        return ZH_min, ZH_max
+
+    def normalize(self):
+        """ Normalize the PDF
+
+        First checks to see if it is normalized
```
Comment (mifumagalli) [inline pyigm/metallicity/pdf.py]: Do we want to check? - probably a good idea for those PDFs with some roundoff errors where the integral is not ==1

**Comment by mifumagalli** [comment]
Quite happy with it - modulo query above. When fixed, happy to merge if JMO approves.

**Comment by astronomeara** [comment]
go for it.  I’m still catching up on the process, but defer to your judgement

File: pyigm/metallicity/pdf.py (hunk 2 above)
Comment (profxj) [inline pyigm/metallicity/pdf.py]: I was going to check and then decided not to (but left
the comment in).  I think it best not to check but just do.

**Comment by profxj** [comment]
Removed the offensive comment.
Pushed.

**Pull request merged**